### DISCUSSION TITLE
Welcome to VulnerableApp Discussions!

### DISCUSSION THREAD

**Discussion by preetkaran20** [SasanLabs/VulnerableApp#280] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nacer-lachhab** [comment]
Hi There,
I want first to report  an issue related to an error that we found on the project, the error is that the package on the class "SampleVulnerability" on the package "resources/sampleVulnerability" is incorrect (due to an coppy past i think), and also i want to ask you about the utility to duplicate it because this class is the same as the class with the same name "SampleVulnerability" on the package "org.sasanlabs.service.vulnerability.sampleVulnerability". to do bref what is the utility to duplicate this class? and is this really an error?

**Reply by preetkaran20** [reply]
Hi @nacer-lachhab ,

Thanks for reaching out to us on github discussions. 
SampleVulnerability is present in resources/sampleVulnerability package and we have deliberately have different package name. The reason behind having a different package name is to serve as a template for setting up newer vulnerabilities in case someone wants to extend our project.

To explain the process further, when a user runs the ./gradlew GenerateSampleVulnerability command, the file is copied from the resources folder and placed in the src/ folder. This copied file is utilized during compilation to create a sample vulnerability, which also becomes visible in the UI. The idea is to help developers who want to extend VulnerableApp quickly get started.

From Readme:
> If you are a developer or a security professional looking to add new Vulnerability type then you can Generate the Sample Vulnerability by running ./gradlew GenerateSampleVulnerability. 
> It will generate the Sample Vulnerability template which has placeholders and comments. Modified files can be seen in the logs of the command or in the github history. You can navigate to those files, fill in the placeholders and then build the project to see the effect of the changes.

Hopefully, It clarifies the doubts of having file in resources and duplicating.
thanks,
Karan
